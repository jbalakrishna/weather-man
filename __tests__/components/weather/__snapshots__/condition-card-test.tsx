import ConditionCard from "@/components/weather/condition-card";
import { render, waitFor } from "@testing-library/react-native";

describe("Condition Card Day", () => {
  test("day mode should render correctly", async () => {
    const tree = render(
      <ConditionCard
        condition={{
          text: "Sunny",
          icon: "icon1",
          code: 120,
          dayColors: ["#fff", "#fff"],
          nightColors: ["#000", "#000"],
        }}
        temp={"23.4"}
        feelsLike={"22.4"}
        isDay
      />
    ).toJSON();

    await waitFor(() => {
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("Condition Card Night", () => {
  test("night mode should render correctly", async () => {
    const tree = render(
      <ConditionCard
        condition={{
          text: "Sunny",
          icon: "icon1",
          code: 120,
          dayColors: ["#fff", "#fff"],
          nightColors: ["#000", "#000"],
        }}
        temp={"23.4"}
        feelsLike={"22.4"}
        isDay={false}
      />
    ).toJSON();

    await waitFor(() => {
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("Condition Card without colors", () => {
  test("without colors should render correctly", async () => {
    const tree = render(
      <ConditionCard
        condition={{
          text: "Sunny",
          icon: "icon1",
          code: 120,
          dayColors: [],
          nightColors: [],
        }}
        temp={"23.4"}
        feelsLike={"22.4"}
        isDay={false}
      />
    ).toJSON();

    await waitFor(() => {
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("Condition Card without icon", () => {
  test("without icon should render correctly", async () => {
    const tree = render(
      <ConditionCard
        condition={{
          text: "Sunny",
          icon: "",
          code: 120,
          dayColors: [],
          nightColors: [],
        }}
        temp={"23.4"}
        feelsLike={"22.4"}
        isDay={false}
      />
    ).toJSON();

    await waitFor(() => {
      expect(tree).toMatchSnapshot();
    });
  });
});
