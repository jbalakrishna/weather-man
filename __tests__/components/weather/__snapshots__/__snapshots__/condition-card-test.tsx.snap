// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condition Card Day day mode should render correctly 1`] = `
<View
  className="flex-row justify-between gap-4 py-4  rounded-xl drop-shadow"
  style={
    {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    className="flex-1 flex-col gap-2"
  >
    <Image
      className="h-40"
      resizeMode="contain"
      source={
        {
          "uri": "icon1",
        }
      }
    />
    <Text
      className="text-center font-bold color-slate-600 text-xl px-4 pb-4"
    >
      Sunny
    </Text>
  </View>
  <View
    className="flex-none gap-8 py-8 pr-8 align-end justify-start"
  >
    <Text
      className="text-right font-bold color-slate-600 text-4xl"
    >
      23.4
    </Text>
    <View>
      <Text
        className="text-right font-semibold color-slate-500 text-xl "
      >
        Feels Like
      </Text>
      <Text
        className="text-right font-bold color-slate-600 text-2xl"
      >
        22.4
      </Text>
    </View>
  </View>
  <Text
    className="absolute bottom-0 right-0 text-right font-bold color-slate-600 text-xl mb-8 mr-8"
  >
    Day
  </Text>
</View>
`;

exports[`Condition Card Night night mode should render correctly 1`] = `
<View
  className="flex-row justify-between gap-4 py-4  rounded-xl drop-shadow"
  style={
    {
      "backgroundColor": "#000",
    }
  }
>
  <View
    className="flex-1 flex-col gap-2"
  >
    <Image
      className="h-40"
      resizeMode="contain"
      source={
        {
          "uri": "icon1",
        }
      }
    />
    <Text
      className="text-center font-bold color-slate-600 text-xl px-4 pb-4"
    >
      Sunny
    </Text>
  </View>
  <View
    className="flex-none gap-8 py-8 pr-8 align-end justify-start"
  >
    <Text
      className="text-right font-bold color-slate-600 text-4xl"
    >
      23.4
    </Text>
    <View>
      <Text
        className="text-right font-semibold color-slate-500 text-xl "
      >
        Feels Like
      </Text>
      <Text
        className="text-right font-bold color-slate-600 text-2xl"
      >
        22.4
      </Text>
    </View>
  </View>
  <Text
    className="absolute bottom-0 right-0 text-right font-bold color-slate-600 text-xl mb-8 mr-8"
  >
    Night
  </Text>
</View>
`;

exports[`Condition Card without colors without colors should render correctly 1`] = `
<View
  className="flex-row justify-between gap-4 py-4  rounded-xl drop-shadow"
  style={
    {
      "backgroundColor": "#E2E8F0",
    }
  }
>
  <View
    className="flex-1 flex-col gap-2"
  >
    <Image
      className="h-40"
      resizeMode="contain"
      source={
        {
          "uri": "icon1",
        }
      }
    />
    <Text
      className="text-center font-bold color-slate-600 text-xl px-4 pb-4"
    >
      Sunny
    </Text>
  </View>
  <View
    className="flex-none gap-8 py-8 pr-8 align-end justify-start"
  >
    <Text
      className="text-right font-bold color-slate-600 text-4xl"
    >
      23.4
    </Text>
    <View>
      <Text
        className="text-right font-semibold color-slate-500 text-xl "
      >
        Feels Like
      </Text>
      <Text
        className="text-right font-bold color-slate-600 text-2xl"
      >
        22.4
      </Text>
    </View>
  </View>
  <Text
    className="absolute bottom-0 right-0 text-right font-bold color-slate-600 text-xl mb-8 mr-8"
  >
    Night
  </Text>
</View>
`;

exports[`Condition Card without icon without icon should render correctly 1`] = `
<View
  className="flex-row justify-between gap-4 py-4  rounded-xl drop-shadow"
  style={
    {
      "backgroundColor": "#E2E8F0",
    }
  }
>
  <View
    className="flex-1 flex-col gap-2"
  >
    <Image
      className="h-40"
      resizeMode="contain"
      source={
        {
          "uri": "",
        }
      }
    />
    <Text
      className="text-center font-bold color-slate-600 text-xl px-4 pb-4"
    >
      Sunny
    </Text>
  </View>
  <View
    className="flex-none gap-8 py-8 pr-8 align-end justify-start"
  >
    <Text
      className="text-right font-bold color-slate-600 text-4xl"
    >
      23.4
    </Text>
    <View>
      <Text
        className="text-right font-semibold color-slate-500 text-xl "
      >
        Feels Like
      </Text>
      <Text
        className="text-right font-bold color-slate-600 text-2xl"
      >
        22.4
      </Text>
    </View>
  </View>
  <Text
    className="absolute bottom-0 right-0 text-right font-bold color-slate-600 text-xl mb-8 mr-8"
  >
    Night
  </Text>
</View>
`;
